fdbserver/storageserver.actor.cpp

```diff
@@ -10120,8 +10120,14 @@ Future<Void> StorageServerMetrics::waitMetrics(WaitMetricsRequest req, Future<Vo
 
 ACTOR Future<Void> waitMetricsTenantAware_internal(StorageServer* self, WaitMetricsRequest req) {
 	if (req.tenantInfo.present() && req.tenantInfo.get().tenantId != TenantInfo::INVALID_TENANT) {
-		wait(success(waitForVersionNoTooOld(self, latestVersion)));
-		Optional<TenantMapEntry> entry = self->getTenantEntry(latestVersion, req.tenantInfo.get());
+		state Version version = wait(waitForVersionNoTooOld(self, latestVersion));
```

Comment:
Discussed offline with A.J.. Here's a summary for anyone else interested:
When `waitForVersionNoTooOld()` is called with `latestVersion` as the argument, the function sets the `version` to smaller of `Version(1)` [1] or the available version in the local `data` [2]. Case [2] wouldn't lead to `future_version` error. Case [1] wouldn't either because the version is set upon restoring from disk and when starting up a new SS in the `storageServer()` functions, before `storageServerCore()`. Also empirically checked that `data->version` is always `>= 1` by adding an assertion and running 100,000 simulation tests.

Reply:
Since we're relying on this behavior now, we should think about adding the assert and making the related changes to the function based on us not needing to handle the case where version < 1. It could happen in a separate PR, though, if we wanted.

Reply:
Update: it turns out that the version on the storage server can indeed be 0 in rare edge cases (<1 in 100,000). I've submitted https://github.com/apple/foundationdb/pull/8766 to handle the `future_version()` error in that case.